src/net/netdevice.rs

```diff
@@ -0,0 +1,84 @@
+//! Low-level Linux network device access
+//!
+//! # References
+//! - [Linux]
+//!
+//! [Linux]: https://man7.org/linux/man-pages/man7/netdevice.7.html
+
+#[cfg(feature = "alloc")]
+use crate::alloc::string::String;
+use crate::fd::OwnedFd;
+use crate::io;
+use crate::io::Errno;
+use crate::net::{socket, AddressFamily, SocketType};
+
+/// Creates a socket used to communicate with the kernel in the ioctl calls.
+#[cfg(target_os = "linux")]
```

Comment:
Since this whole module is conditional on the target OS, it's not necessary to add `cfg`s like this to individual items.

Reply:
Done.